staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/plugin.go

```diff
@@ -72,5 +74,10 @@ func (a *Plugin) ValidateInitialization() error {
 
 // Admit makes an admission decision based on the request attributes.
 func (a *Plugin) Admit(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	wd, ok := request.WebhookDurationFrom(ctx)
+	if ok {
+		admitStart := time.Now()
+		defer func() { wd.SetAdmit(time.Since(admitStart)) }()
+	}
```

Comment:
Also - I would actually prefer to go deeper than this. So really count only the calls to webhooks and not the latency of the whole webhook machinery. If that machinery is slow for some reason, this is our problem, not the customer problem.

So instead of this, I would actually:
1) change the webhook-duration to expose the "AddAdmit[Duration]" instead of "SetAdmit" method
2) does the time-counting around the actual webhook call here:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go#L266
i.e. change that line into:
```
admitStart := time.Now()
err := r.Do(ctx).Into(response)
wd.AddAdmit(clock.Since(admitStart)
if err != nil {
  // the rest goes here
```

Reply:
maybe we can use some abstraction here since we will need to call it in multiple places
```
type ExternalLatencyTracker interface {
	Track(func())
	GetTotalLatency() time.Duration
}

type tracker struct {
	clock clock.Clock
	latency time.Duration
}

func (t *tracker) Track(f func()) {
	startedAt := t.clock.Now()
	defer func() {
		t.latency += t.clock.Since(startedAt)
	}()

	f()
}

type fakeExternalLatencyTracker struct{}
func (t *fakeExternalLatencyTracker) Track(f func()) {
	f()
}

// we can store the interface in the request context
type externalLatencyTrackerKeyType int
const externalLatencyTrackerKey externalLatencyTrackerKeyType = iota

func ExternalLatencyTrackerFrom(ctx context.Context) ExternalLatencyTracker {
	if tracker, ok := ctx.Value(externalLatencyTrackerKey).(ExternalLatencyTracker); ok {
		return tracker
	}
	return &fakeExternalLatencyTracker{}
}

// where we use it.
var err error
externalLatencyTracker := request.ExternalLatencyTrackerFrom(ctx)
externalLatencyTracker.Track( func() {
	err = r.Do(ctx).Into(response)
})
```

Reply:
Fake isn't really needed - I want to measure time anyway (see my other comment).

I'm fine with introducing this interface, but TBH, I don't think it will simplify a lot..

Reply:
the fake is just to make sure that `ExternalLatencyTrackerFrom` never returns a nil instance although it should never happen in real scenario.

I think I see usefulness of the abstraction is in:
- latency tracking logic is in a single unit, the call sites are more simple now - no need to use defer or use a clock to get the current time.
- we can test the `Track` function in isolation - make sure latency is added as expected, make sure we see expected behavior if the function called panics (otherwise, we have to exercise these test cases in N call sites)
- the tests for the call sites, all we need to do is use a stubbed `ExternalLatencyTracker` that just makes sure that the Track function is invoked, no extra verification step is required since we have already verified that `Track` function works as expected.

Reply:
Thank you for suggestions. I've moved time measurement to dispatch. Added tracker abstraction but I believe adding fake tracker would make it less transparent.
About not happening in real scenario, If I remember correctly from some testing I've done I think readyz or some other health check request doesn't go though DefaultBuildHandlerChain so I believe for some requests webhook duration may be not initialized in context.